File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.py
def solution(n):
    answer = 0
    F = [0] * (n+1)
    F[0], F[1] = 0, 1
    
    for i in range(2,n+1):
        F[i] = F[i-1] + F[i-2]
    
    return F[-1] % 1234567
    

File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n) {
    vector<int> dp(n+1);
    dp[0] = 0;
    dp[1] = 1;
    for (int i = 2; i<=n; i++){
        dp[i] = (dp[i-1] + dp[i-2])%1234567;
    }
    // printf(dp);
    int answer = dp[n];
    return answer;
}